UnitySDK/Assets/ML-Agents/Scripts/Academy.cs

```diff
@@ -469,16 +469,27 @@ public int GetEpisodeCount()
         }
 
         /// <summary>
-        /// Returns the current step counter (within the current epside).
+        /// Returns the current step counter (within the current episode).
         /// </summary>
         /// <returns>
-        /// Current episode number.
+        /// Current step count.
         /// </returns>
         public int GetStepCount()
         {
             return stepCount;
         }
 
+        /// <summary>
+        /// Returns the total step counter.
+        /// </summary>
+        /// <returns>
+        /// Total step count.
+        /// </returns>
+        public int GetTotalStepCount()
+        {
+            return GetStepCount() + GetEpisodeCount() * maxSteps;
```

Comment:
Looks like this makes the assumption that every episode reaches `maxSteps`.  In reality we don't have any current account of how many steps each historical episode lasted.  We'll have to store total step count for this method to work correctly.

Reply:
I am not sure that this is correct, the variable names are actually confusing here. I think the maxSteps here is the number of steps between academy resets and the episode count is the number of academy episodes not agent episodes.

The academy should only reset when it reaches its max step right?

But after checking it seems that there is another problem anyway. It looks like the academy is resetting twice every time and the episode count is also incremented twice.

So in the end, I added a variable for totalStepCount, it makes it easier to understand the code anyway.

It should be fixed in b0d1921 👍

Reply:
I think that's mostly correct except that you might reset the Academy early (for example by explicitly calling the Gym wrapper's `reset()`).  Anyway, this change looks good to me :)